cmd/server-rlimit.go

```diff
@@ -39,6 +46,21 @@ func setMaxResources() (err error) {
 	return err
 }
 
+// increaseMaxThreads sets a new number of maximum allowed
+// go routines if it is greater than the old one.
+func increaseMaxThreads(max int) int {
+	if max <= 0 {
+		return max
+	}
+	// There is no way to know the current number of max threads but
+	// only after setting the new number. If the latter turned out
+	// to be less than the old one, revert the operation.
+	if oldMax := debug.SetMaxThreads(max); oldMax > max {
+		return debug.SetMaxThreads(oldMax)
```

Comment:
As per https://golang.org/pkg/runtime/debug/#SetMaxThreads, the default value is 10,000 threads. If max value is < 10,000, you reset back to 10,000 threads.

In the above `maxLimit` of `sys.GetMaxOpenFileLimit()` is used which is 4096 in Linux by default.   In this case, the function sets back 10,000 again.

I think this function doesn't work practically.

Have you tried with other *nix?